src/app/onion/learning-object-builder/components/outcome/outcome-typeahead/outcome-typeahead.component.ts

```diff
@@ -141,22 +141,33 @@ export class OutcomeTypeaheadComponent implements OnInit, OnChanges, OnDestroy {
    * @return {boolean} true if verb is found, false otherwise
    */
   isGoodVerb(verb: string, noEmit?: boolean): boolean {
+    if (!this.bloom) {
+      const entries = Object.entries(taxonomy.taxons);
+      const match = entries.find(([_, taxon]) => taxon.verbs.includes(verb.toLowerCase()));
+
+      if (match) {
+        this.bloom = match[0];
+      }
+    }
+
     const levelsArray = Array.from(levels.values());
     for (let i = 0, l = levelsArray.length; i < l; i++) {
-      // for this level, grab it's verbs and check if the current verb is in that list
-      if (taxonomy.taxons[levelsArray[i]].verbs.includes(verb.toLowerCase())) {
-        this.bloom = levelsArray[i];
-        if (!noEmit) {
-          this.selectedCategory.emit(this.bloom);
+      if (levelsArray[i] === this.bloom) {
+        if (taxonomy.taxons[levelsArray[i]].verbs.includes(verb.toLowerCase())) {
+          this.bloom = levelsArray[i];
+          if (!noEmit) {
+            this.selectedCategory.emit(this.bloom);
+          }
+          return true;
         }
```

Comment:
all that needed to be added was an extra check to make sure this.bloom (the selected category) matches one of the levels in the taxonomy, but then another if block was needed at the start of the function to handle a bloom not being selected from the drop down in the scenario that an outcome is copy and pasted in. In that case, we only have the verb which we use to search up through the taxons structure to find the key and if it is a duplicate verb, the first category that contains it will be assigned to this.bloom